### DISCUSSION TITLE
Ignore punctuation marks while offering suggestions ?

### DISCUSSION THREAD

**Discussion by ghost** [OpenBangla/OpenBangla-Keyboard#361] [Ideas]
## "আমি পৌছাইসি Barcode Cafe, তুমি কোথায় ?"

When typing "Cafe", OBK turns it into "চাফে", I manually selected "Cafe" from the suggestions list, then I typed a comma, so what I typed became "Cafe," , OBK turns it into "চাফে,", basically undoing the previous manual selection of "Cafe".

![1](https://github.com/OpenBangla/OpenBangla-Keyboard/assets/138502262/82454012-2f9e-4593-a626-4a327c5beadd) ![2](https://github.com/OpenBangla/OpenBangla-Keyboard/assets/138502262/d8934521-485c-4fe2-bc6c-ac5623fc7e1f) ![3](https://github.com/OpenBangla/OpenBangla-Keyboard/assets/138502262/c6dd3292-42a6-4d59-926a-1e9aa5a9be36)

Also found another bug/issue(?) (it suggests double quotes twice)
![4](https://github.com/OpenBangla/OpenBangla-Keyboard/assets/138502262/c6f9c158-45a7-4630-a39a-2a1060e24007)

This is a minor issue and might sound nitpicky. But I am a moderately fast typer and when this happens several times while writing it becomes very distracting and annoying.
AFAIK punctuation like commas, semi-colons, fullstops don't influence how Bangla is phoenetically displayed/typed.
Thus, I believe OBK's punctuation mark handling can be improved upon.

## My idea of a fix :
A good way to handle this could be to treat punctuation marks the same as pressing "Space"
So when a user types a comma/semicolon/quotation mark after a word, the prediction/suggestion stops and OBK moves onto the next word, like how pressing "Space" would do.

**Comment by mominul** [comment]
I have implemented it, can you have a check? @larina3315 

You can run the following commands in the directory you have cloned OBK to have the updates:
```
git pull
git submodule update
cd build
sudo make install
```

> Also found another bug/issue(?) (it suggests double quotes twice)

This also has been fixed now!

**Reply by ghost** [reply]
Sure thing, I will update it asap and let you know.
Thank you for your hardwork.

**Reply by ghost** [reply]
I have tested it on Fedora Kinoite 39 (uBlue kinoite-main image)  (39.20231223.0)
It works fine, the issues mentioned seem to have been fixed, I will test it more over the next couple of days, just in case for any regressions/instability.

**Reply by mominul** [reply]
Would you please try out the ibus engine also?

**Reply by ghost** [reply]
I have tried to install a ibus OBK 3 rpm i just compiled with the -DENABLE_IBUS=ON flag
but it errors out with `error: Importing package 'ibus-openbangla': Analyzing /usr/local: Unsupported path; see https://github.com/projectatomic/rpm-ostree/issues/233`

It appears that immutable Fedora spins don't like rpm files writing to /usr/local and /opt
Also, I did a `git pull` before compiling and noticed some additions/deletions, was there any recent packaging-related changes ?
Because so far I have used ibus OBK 2/3 and fcitx OBK 3 on Silverblue/Kinoite without any issues.

**Reply by ghost** [reply]
I compiled another package, this time with that extra flag on, it installed fine.
Tried it on Silverblue with the ibus OBK 3 package, the punctuation mark suggestion fix works perfectly fine

**Reply by mominul** [reply]
Thanks! :heart:

**Comment by ghost** [comment]
closing as solved
